operator/pkg/controllers/etcd/pod.go

```diff
@@ -187,10 +169,23 @@ func podSpecFor(controlPlane *v1alpha1.ControlPlane) *v1.PodSpec {
 	}
 }
 
+func generateEtcdBootstrapCmd(controlPlane *v1alpha1.ControlPlane) string {
+	return fmt.Sprintf(etcdBootstrapCmd, etcdctlEndpoints(controlPlane), advertizePeerURL(controlPlane),
+		initialClusterFlag(controlPlane), advertizeClusterURL(controlPlane))
+}
+
 func initialClusterFlag(controlPlane *v1alpha1.ControlPlane) string {
+	return endPointFlagGenerator(controlPlane, "%[1]s-etcd-%[2]d=https://%[1]s-etcd-%[2]d.%[1]s-etcd.%[3]s.svc.cluster.local:2380")
+}
+
+func etcdctlEndpoints(controlPlane *v1alpha1.ControlPlane) string {
+	return endPointFlagGenerator(controlPlane, "https://%[1]s-etcd-%[2]d.%[1]s-etcd.%[3]s.svc.cluster.local:2379")
+}
+
+func endPointFlagGenerator(controlPlane *v1alpha1.ControlPlane, format string) string {
 	nodes := make([]string, 0)
 	for i := 0; i < controlPlane.Spec.Etcd.Replicas; i++ {
-		nodes = append(nodes, fmt.Sprintf("%[1]s-etcd-%[2]d=https://%[1]s-etcd-%[2]d.%[1]s-etcd.%[3]s.svc.cluster.local:2380", controlPlane.ClusterName(), i, controlPlane.Namespace))
+		nodes = append(nodes, fmt.Sprintf(format, controlPlane.ClusterName(), i, controlPlane.Namespace))
```

Comment:
Usually we pass in order of (namespace, name), since resources are denoted this way in k8's paths (e.g. parent/child in rest).